transitiveDeps.md

```diff
@@ -0,0 +1,94 @@
+# Dependency Management
+
+The Azure SDK for JS client libraries include a host of dependencies, both internal and external. In general, our policy allows for some flexibility in dependencies and we generally pin to a [caret][caret] version - allowing for patches, security fixes, and any non-breaking changes in our dependency ranges.
+
+There are times when a dependency needs to be updated, such as when a security vulnerability has been found and patched in the depdendent codebase. While we will address security vulnerabilities in dependencies by updating our minimum version to the latest patched version, our semver policy allows customers to update to the latest version of a transitive dependency without waiting for our next releases and without requiring a hotfix. In this document we'll outline a few options available to you when a transitive dependency must be updated.
+
+## Example scenario
+
+Let's assume that a vulnerability has been found in [node-fetch][node-fetch] version 2.6.6 and that you use `@azure/keyvault-keys` directly. Your security scan identified this vulnerability and asks to upgrade to 2.6.7 or higher.
+
+The outputs below assume you're using `npm v8.1.2` with a lockfile, but we will provide instructions for Yarn as well.
+
+### Identifying the dependency tree
+
+First, let's see how `node-fetch` is pulled into our dependency tree.
+
+```bash
+keyvault main % npm ls node-fetch
+keyvault@1.0.0 /home/user/my-app
+└─┬ @azure/keyvault-keys@4.3.0
+  └─┬ @azure/core-http@2.2.3
+    └── node-fetch@2.6.6
+```
+
+> If you are using Yarn you can use `yarn why node-fetch` with similar results.
+
+It looks like `node-fetch`, a dependency of `@azure/core-http@2.2.3` is being pulled in _transitively_ via `@azure/keyvault-keys@4.3.0`.
+
+Because I use a lockfile, running `npm install` again will not help me here. But `npm` provides a few utilities that can.
+
+### Using `npm audit fix`
+
+First, it's possible that you were alerted to a security vulnerability thanks to `npm audit`. In this case, the simplest solution might be to run `npm audit fix`. Let's see what that looks like:
+
+```bash
+keyvault main % npm audit fix
+
+changed 1 package, and audited 51 packages in 421ms
+
+2 packages are looking for funding
+  run `npm fund` for details
+
+found 0 vulnerabilities
+keyvault main % npm ls node-fetch
+keyvault@1.0.0 /home/user/my-app
+└─┬ @azure/keyvault-keys@4.3.0
+  └─┬ @azure/core-http@2.2.3
+    └── node-fetch@2.6.7
+```
+
+As you can see node-fetch has been updated to 2.6.7, without having to wait for a new version of `@azure/core-http`.
+
+For more information on `npm audit` please refer to the [npm-audit documentation][npm-audit].
+
+> If you are using Yarn you can use [yarn npm audit][yarn-npm-audit] with similar results.
```

Comment:
[yarn npm audit] is not declared below, so the link is not working here.

Reply:
Thanks Jonathan! I manually applied all of your suggestions so will be resolving these comments. Thanks for reviewing!